File: BinarySearch.py
lis=[]   # creating the list
num=int(input("enter the number of elements you want to enter"))
for i in range(num):
    num1=int(input("enter element"))
    lis.append(num1)
beg=0
end=num
find=int(input("enter the element you want to search"))
while(beg<end):
    mid=(beg+end)//2  # finding the midle element
    if(find<lis[mid]))  #checking the condition of smaller number
        end=mid-1
    elif find>lis[mid]:  # checking the condition of greater number
        beg=mid+1
    elif find == lis[mid]:  3the number is found
        pos=mid
        print("the element is found a position", pos)
        break
    

File: Binary_search.c
#include<stdio.h>
#include<conio.h>
void main()
{
  int arr[50],i,n,loc=0,beg=0,end,mid,item;
  clrscr();
  printf("ENTER NUMBER OF ELEMENT\n");
  scanf("%d",&n);
  end=n-1;
  printf("ENTER ELEMENT IN SORTED ORDER\n");
  for(i=0;i<n;i++)
    scanf("%d",&arr[i]);
  printf("ENTER ELEMENT TO BE SEARCHED\n");
  scanf("%d",&item);
  while(beg<=end)
   {
     mid=((beg+end)/2);
     if(arr[mid]<item)
       beg=mid+1;
     else if(arr[mid]==item)
     {
       printf("%d FOUND AT LOCATION %d.\n",item,mid+1);
       break;
     }
     else
       end=mid-1;
     mid=(beg+end)/2;
   }

   if(beg>end)
     printf("ITEM NOT FOUND!!");
   getch();
}
